Add Management SDK for Microsoft.DevCenter

tbyfield opened the pull request:
Adding mangement SDK Azure.ResourceManager.DevCenter for the Microsoft.DevCenter RP - this is currently in public preview.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

tbyfield committed:
Add Management SDK for Microsoft.DevCenter

tbyfield committed:
Update Test Environment variables to capitalize names

ArthurMa1978 commented on sdk/devcenter/ci.yml:
```diff
@@ -0,0 +1,31 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
```
we have updated the ci.yml to ci.mgmt.yml, the also change the setting, you can refer https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/avs/ci.mgmt.yml

tbyfield replied:
Renamed and updated the file

ArthurMa1978 commented on sdk/devcenter/Azure.ResourceManager.DevCenter/README.md:
```diff
@@ -0,0 +1,76 @@
+# Azure DevCenter Management client library for .NET
```
`DevCenter` to `Dev Center`

ArthurMa1978 commented on sdk/devcenter/Azure.ResourceManager.DevCenter/README.md:
```diff
@@ -0,0 +1,76 @@
+
+### Install the package
+
+Install the Azure DevCenter management library for .NET with [NuGet](https://www.nuget.org/):
```
`DevCenter` to `Dev Center`

ArthurMa1978 commented on sdk/devcenter/Azure.ResourceManager.DevCenter/package-lock.json:
```diff
@@ -0,0 +1,11 @@
+{
```
remove this file

tbyfield replied:
Removed

ArthurMa1978 commented on sdk/devcenter/Azure.ResourceManager.DevCenter/src/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,9 @@
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("DevCenter")]
```
Microsoft.DevCenter

tbyfield replied:
done

tbyfield committed:
PR comment changes:
- rename "DevCenter" to "Dev Center" in readme
- Update AzureResourceProviderNamespace
- remove package lock file

tbyfield committed:
Updated/renamed ci.yml -> ci.mgmt.yml

tbyfield commented:
@ArthurMa1978 thanks for the feedback, I've updated as per your comments.

ArthurMa1978 commented:
/azp run prepare-pipelines

ArthurMa1978 commented:
/azp run net - devcenter - mgmt

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.DevCenter](https://apiview.dev/Assemblies/Review/dd9eba197b5c4e5e9937e17dbf168370)

ArthurMa1978 commented:
Please run the `eng\scripts\Update-Snippets.ps1 devcenter` to generate the apiview file

tbyfield committed:
Add apiview file

ArthurMa1978 commented:
/azp run net - devcenter - mgmt

ArthurMa1978 commented:
@tbyfield just realized there is a new process run, please run `eng/scripts/Update-Mgmt-CI.ps1` to update the ci setting for a new sdk

tbyfield committed:
Merge branch 'main' into devcenter-mgmt-2022-08-01-preview

tbyfield committed:
Ran Update-Mgmt-CI.ps1 to add DevCenter to ci.mgmt.yml
